lib/lrama/grammar.rb

```diff
@@ -507,6 +508,21 @@ def fill_sym_to_rules
       end
     end
 
+    # @rbs () -> void
+    def validate_precedence_is_term!
+      errors = [] #: Array[String]
+
+      @rules.each do |rule|
```

Comment:
I feel it makes the intention more clear if we iterate over the `@nterms` even so all of non terminals should appear on the some rule's LHS.